core/execution/future/market.go

```diff
@@ -5598,6 +5597,14 @@ func (m *Market) AmendAMM(ctx context.Context, amend *types.AmendAMM, determinis
 	if order == nil {
 		m.amm.Confirm(ctx, pool)
 		m.matching.UpdateAMM(pool.AMMParty)
+		if pool.Parameters.LowerBound == nil || pool.Parameters.UpperBound == nil {
+			// pool has switched sides, or we no longer are quoting one side
+			if (existing.Parameters.LowerBound == nil && pool.Parameters.LowerBound != nil) ||
+				(existing.Parameters.UpperBound == nil && pool.Parameters.UpperBound != nil) ||
+				(existing.Parameters.UpperBound != nil && existing.Parameters.LowerBound != nil) {
+				m.checkForReferenceMoves(ctx, nil, true)
+			}
+		}
```

Comment:
not a big deal, but I'd put these condition on a method of the pool maybe if possible, just because if we need to update it here, we may miss the other bit. but not a blocker.